test/Chainweb/Test/CutDB.hs

```diff
@@ -111,8 +120,8 @@ withTestCutDb
     -> LogFunction
         -- ^ a logg function (use @\_ _ -> return ()@ turn of logging)
     -> (forall cas . PayloadCasLookup cas => CutDb cas -> IO a)
-    -> IO a
-withTestCutDb rdb v n pactIO logfun f = do
+    -> IO (RocksDbCas CutHashes, a)
```

Comment:
Why is this needed. It is probably safe to let the cut-hashes table escape the scope of `withTestCutDb`, but is a bit unintuitive. For instance, some future code refactoring may implement some sort of cleanup for the temporary test cut db, which would cause code using the result db to break.

Reply:
I'm not sure I follow; we can't access the `cutHashesStore` from the `CutDb` record.

Reply:
I don't understand. Can't the whole test, including the assertions be moved into the `withTestCutDb` bracket?

If `cutHashesStore` isn't available from `CutDb`, the type of the inner action could become:

```Haskell
(forall cas . PayloadCasLookup cas => CutDb cas -> RocksDbCas CutHashes -> IO a)
```

That way any access to the cutDb and the cutHashesStore would occur within the bracket.

Reply:
I had the same thought this morning, implemented it, and just read this now :D